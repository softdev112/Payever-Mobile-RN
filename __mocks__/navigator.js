export default {
  setOnNavigatorEvent: jest.fn(),
  resetTo: jest.fn(),
  push: jest.fn(),
  pop: jest.fn(),
  dismissLightBox: jest.fn(),
  dismissModal: jest.fn(),
  showModal: jest.fn(),
};