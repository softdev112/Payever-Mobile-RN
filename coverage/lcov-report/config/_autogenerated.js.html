<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for config/_autogenerated.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">config</a> _autogenerated.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/0</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">/* eslint-disable max-len */
export default {
  siteUrl: 'https://stage.payever.de',
  api: {
    baseUrl: 'https://stage.payever.de',
    clientId: '117_5sd54ii2lyo8sg84g0w4w08wkw4040gks4k0sk0wgc44w48404',
    clientSecret: '4jsu5lynawsg8g4cs80w4kcsocwcggwo048ogs8c844gk80w80',
  },
&nbsp;
  oauthData: {
    twitter: {
      appId: 'oRCkHDAnAFSXiO1wayGlpBkwn',
      appSecret: 'C3FUgIBnjdn1Je6dD5NX5AboFEJ1vWqVVwOLJ03Itk0Xp58but',
      callback: '/social/auth/twitter',
    },
&nbsp;
    facebook: {
      appId: '306625799760428',
      appSecret: '4e0e5079b5c7eeff9f16295f0212ca95',
      callback: 'fb306625799760428://authorize',
      scope: 'email public_profile',
      fields: [
        'id', 'name', 'first_name', 'last_name',
        'picture.type(large)', 'email', 'cover', 'birthday',
      ],
    },
&nbsp;
    google: {
      appId: '463919467171-u54vacd9oiso7l14f97g38odsak91njg.apps.googleusercontent.com',
      appSecret: 'SihRvMNWT2j5088wEZ_A2MRR',
      callback: 'com.googleusercontent.apps.463919467171-u54vacd9oiso7l14f97g38odsak91njg:/google',
      scope: 'https://www.googleapis.com/auth/content',
    },
&nbsp;
    linkedIn: {
      appId: '78a1v96xm7x1tt',
      appSecret: 'Umu6R25x1w7NiDt6',
      callback: 'https://localhost:3003/auth/linkedin',
      scope: 'r_basicprofile r_emailaddress',
    },
  },
};</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Jul 06 2017 09:31:11 GMT+0300 (MSK)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
