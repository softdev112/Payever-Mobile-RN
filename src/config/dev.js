/* eslint-disable max-len */
import local from './local';

export default Object.assign({
  siteUrl: 'https://mein.payever.de',
  api: {
    baseUrl: 'https://mein.payever.de',
    clientId: '1633_3vi6g4uiwmyow8044o4w0wo4s88cogosw84kw888kw408wok8c',
    clientSecret: '63nqx0kppzocw4sswok8gg800sk4w4kko4k0oo8gs4ow00wgos',
  },
  debug: {
    logApiCall: true,
  },

  oauthData: {
    twitter: {
      appId: 'oRCkHDAnAFSXiO1wayGlpBkwn',
      appSecret: 'C3FUgIBnjdn1Je6dD5NX5AboFEJ1vWqVVwOLJ03Itk0Xp58but',
      callback: '/social/auth/twitter',
    },

    facebook: {
      appId: '438908279593357',
      appSecret: 'cd96a62e92a5bc498896209613e2d0bf',
      callback: '/social/auth/facebook',
      scope: 'email public_profile',
      fields: [
        'id', 'name', 'first_name', 'last_name',
        'picture.type(large)', 'email', 'cover', 'birthday',
      ],
    },

    google: {
      appId: '463919467171-u54vacd9oiso7l14f97g38odsak91njg.apps.googleusercontent.com',
      appSecret: 'SihRvMNWT2j5088wEZ_A2MRR',
      callback: 'com.googleusercontent.apps.463919467171-u54vacd9oiso7l14f97g38odsak91njg:/google',
      scope: 'https://www.googleapis.com/auth/content',
    },

    linkedIn: {
      appId: '78a1v96xm7x1tt',
      appSecret: 'Umu6R25x1w7NiDt6',
      callback: 'https://localhost:3003/auth/linkedin',
      scope: 'r_basicprofile r_emailaddress',
    },
  },
}, local);