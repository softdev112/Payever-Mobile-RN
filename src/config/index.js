import generatedConfig from './_autogenerated';

export class Config {
  siteUrl: string;
  api: {
    baseUrl: string;
    clientId: string;
    clientSecret: string;
  };
  debug: {
    consoleLevel: boolean;
    logApiCall: boolean;
    logMobx: boolean;
    logSLevel: boolean;
    logSUrl: string;
    logWampCall: boolean;
  };

  constructor(data) {
    data.debug = data.debug || {};
    Object.assign(this, data);
  }
}

export default new Config(generatedConfig);