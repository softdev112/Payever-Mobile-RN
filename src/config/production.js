/* eslint-disable max-len */
export default {
  siteUrl: 'https://mein.payever.de',
  // It's better to prevent using clientSecret. But after analyzing a lot
  // of resources I've made a decision that is's impossible.
  // FOSOAuthServerBundle doesn't support password auth without client_secret
  // https://github.com/FriendsOfSymfony/FOSOAuthServerBundle/issues/115
  api: {
    baseUrl: 'https://mein.payever.de',
    clientId: '1633_3vi6g4uiwmyow8044o4w0wo4s88cogosw84kw888kw408wok8c',
    clientSecret: '63nqx0kppzocw4sswok8gg800sk4w4kko4k0oo8gs4ow00wgos',
  },
  reqTimeout: 10000,

  oauthData: {
    twitter: {
      appId: 'oRCkHDAnAFSXiO1wayGlpBkwn',
      appSecret: 'C3FUgIBnjdn1Je6dD5NX5AboFEJ1vWqVVwOLJ03Itk0Xp58but',
      callback: '/social/auth/twitter',
    },

    facebook: {
      appId: '302838276848223',
      appSecret: '188dcf1590b667929d5f8b4538b00075',
      callback: 'fb302838276848223://authorize',
      scope: 'email public_profile',
      fields: [
        'id', 'name', 'first_name', 'last_name',
        'picture.type(large)', 'email', 'cover', 'birthday',
      ],
    },

    google: {
      appId: '463919467171-u54vacd9oiso7l14f97g38odsak91njg.apps.googleusercontent.com',
      appSecret: 'SihRvMNWT2j5088wEZ_A2MRR',
      callback: 'com.googleusercontent.apps.463919467171-u54vacd9oiso7l14f97g38odsak91njg:/google',
      scope: 'https://www.googleapis.com/auth/content',
    },

    linkedIn: {
      appId: '78a1v96xm7x1tt',
      appSecret: 'Umu6R25x1w7NiDt6',
      callback: 'https://localhost:3003/auth/linkedin',
      scope: 'r_basicprofile r_emailaddress',
    },
  },
};