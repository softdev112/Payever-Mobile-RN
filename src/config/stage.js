export default {
  siteUrl: 'https://stage.payever.de',
  api: {
    baseUrl: 'https://stage.payever.de',
    clientId: '117_5sd54ii2lyo8sg84g0w4w08wkw4040gks4k0sk0wgc44w48404',
    clientSecret: '4jsu5lynawsg8g4cs80w4kcsocwcggwo048ogs8c844gk80w80'
  }
}