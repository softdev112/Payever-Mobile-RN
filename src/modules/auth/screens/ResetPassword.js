import { Component } from 'react';
import { inject, observer } from 'mobx-react/native';
import { Navigator, Navigation } from 'react-native-navigation';
import {
  Button, Container, Error, Loader, NavBar, StyleSheet, TextInput, View,
} from 'ui';

import AuthStore from '../../../store/auth';

/* eslint-disable max-len */
const RESP_SUCCESS_MSG = 'Password was reset successfully. Checkout your email please.';
const RESP_ERROR_MSG = 'There were some errors while resetting password. Try it later please.';
const EMAIL_VALID_PATTERN = '^([a-z0-9_-]+\\.)*[a-z0-9_-]+@[a-z0-9_-]+(\\.[a-z0-9_-]+)*\\.[a-z]{2,6}$';

const errorMessage = 'It should be not empty valid email address for example someone@gmail.de';
/* eslint-enable max-len */

@inject('auth')
@observer
export default class ResetPassword extends Component {
  static navigatorStyle = {
    navBarHidden: true,
  };

  props: {
    auth: AuthStore;
    navigator: Navigator;
  };

  state: {
    email: string;
  };

  $input: TextInput;

  constructor(props) {
    super(props);

    this.state = {
      email: '',
    };
  }

  componentDidMount() {
    // To avoid blinking while screen push animation going on
    if (this.$input) {
      setTimeout(() => this.$input.focus(), 500);
    }
  }

  async onResetPassword() {
    const { auth, navigator } = this.props;
    const { email } = this.state;

    if (email === '' || !email.match(EMAIL_VALID_PATTERN)) {
      auth.setError(errorMessage);
      return;
    }

    const isSuccess = await auth.resetPassword(email);
    const message = isSuccess ? RESP_SUCCESS_MSG : RESP_ERROR_MSG;

    Navigation.showInAppNotification({
      screen: 'core.PushNotification',
      title: '',
      passProps: {
        message,
      },
      navigatorStyle: {
        screenBackgroundColor: 'rgba(0,0,0,0.0)',
      },
    });

    navigator.pop({ animated: true });
  }

  render() {
    const { auth } = this.props;
    const { email } = this.state;

    return (
      <View style={styles.container}>
        <NavBar>
          <NavBar.Back title="Login" showTitle="always" />
          <NavBar.Title title="Reset Password" />
        </NavBar>
        <Loader isLoading={auth.isLoading}>
          <Container contentContainerStyle={styles.form} layout="small">
            <Error
              onShowEnd={() => auth.setError('')}
              duration={5000}
              message={auth.error}
            />
            <View>
              <TextInput
                ref={ref => this.$input = ref}
                autoCapitalize="none"
                autoCorrect={false}
                keyboardType="email-address"
                label="Email address"
                onChangeText={text => this.setState({ email: text })}
                returnKeyType="next"
                value={email}
              />
            </View>
            <View style={styles.submitContainer}>
              <Button
                animated
                disabled={!email.match(EMAIL_VALID_PATTERN)}
                title="Resset Password"
                onPress={::this.onResetPassword}
              />
            </View>
          </Container>
        </Loader>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'column',
    flex: 1,
  },

  header: {
    justifyContent: 'flex-start',
  },

  form: {
    flexDirection: 'column',
    marginTop: '10%',
  },

  submitContainer: {
    marginTop: 20,
  },
});