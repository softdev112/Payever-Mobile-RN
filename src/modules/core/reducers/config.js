import { Map } from 'immutable';
import config from '../../../config';

const initialState = Map(config);

/**
 * Share config through store
 */
export default function configReducer(state = initialState) {
  return state;
}