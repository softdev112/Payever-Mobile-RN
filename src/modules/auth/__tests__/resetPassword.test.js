/* eslint-disable max-len */
import React from 'react';
import renderer from 'react-test-renderer';
import ResetPassword from '../screens/ResetPassword';
import Store from '../../../store';
import config from '../../../config';
import navigator from '../../../../__mocks__/navigator';

jest.mock('../../../common/ui/FlatTextInput', () => 'FlatTextInput');

describe('modules/auth/screens/ResetPassword', () => {
  let store;
  let auth;

  beforeAll(() => {
    ResetPassword.prototype.getChildContext = function getContext() {
      return { navigator };
    };

    ResetPassword.childContextTypes = {
      navigator: React.PropTypes.object,
    };
  });

  beforeEach(() => {
    store = new Store(config);
    auth = store.auth;
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('ResetPassword should render correctly if', () => {
    const tree = renderer.create(
      <ResetPassword
        auth={auth}
        navigator={navigator}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});