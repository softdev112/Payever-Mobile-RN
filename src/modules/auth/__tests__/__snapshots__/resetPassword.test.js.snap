// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modules/auth/screens/ResetPassword ResetPassword should render correctly if 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      Array [
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFF",
            "borderBottomColor": "#e1e1e1",
            "borderBottomWidth": 1,
            "elevation": 8,
            "flexDirection": "row",
            "height": 55,
            "justifyContent": "flex-end",
            "marginTop": 15,
            "paddingHorizontal": 18,
            "zIndex": 1000,
          },
        ],
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "bottom": 0,
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "left": 18,
            "position": "absolute",
            "top": 0,
            "zIndex": 1002,
          },
          null,
        ]
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        collapsable={undefined}
        hitSlop={
          Object {
            "bottom": 10,
            "left": 10,
            "right": 10,
            "top": 10,
          }
        }
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "opacity": 1,
            "padding": 6,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          source="icon-arrow-left-ios-24"
          style={
            Array [
              Object {
                "fontFamily": "payeverIcons",
                "fontSize": 24,
              },
              Array [
                Object {
                  "fontFamily": "payeverIcons",
                  "fontSize": 24,
                },
                Array [
                  Object {
                    "height": 24,
                    "width": 24,
                  },
                  Object {
                    "color": "#0084ff",
                  },
                  undefined,
                ],
              ],
            ]
          }
          touchStyle={
            Object {
              "alignItems": "center",
              "borderColor": "blue",
              "borderWidth": 1,
              "justifyContent": "center",
            }
          }
        >
          Ó®ê
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#0084ff",
                "fontSize": 16,
                "marginLeft": 2,
              },
              Object {
                "fontSize": 16,
                "fontWeight": "400",
                "marginLeft": 2,
              },
            ]
          }
        >
          Login
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 0,
          "flexDirection": "row",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 1001,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        collapsable={undefined}
        hitSlop={
          Object {
            "bottom": 10,
            "left": 10,
            "right": 10,
            "top": 10,
          }
        }
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
            "padding": 6,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000",
                "fontSize": 18,
                "textAlign": "center",
              },
              Object {},
            ]
          }
        >
          Reset Password
        </Text>
      </View>
    </View>
    <Image
      source={1}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": null,
          "justifyContent": "flex-end",
          "paddingLeft": 20,
          "resizeMode": "stretch",
          "width": null,
          "zIndex": 1002,
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "column",
        },
        undefined,
      ]
    }
  >
    <RCTScrollView
      contentContainerStyle={
        Object {
          "flexDirection": "column",
          "marginTop": 133.4,
        }
      }
      keyboardShouldPersistTaps="always"
      layout="small"
      style={
        Array [
          Object {
            "maxWidth": 400,
            "width": 615,
          },
          undefined,
        ]
      }
    >
      <View>
        <View>
          <View
            style={
              Array [
                Object {
                  "height": 60,
                  "paddingBottom": 4,
                  "paddingTop": 28,
                  "position": "relative",
                },
                undefined,
              ]
            }
          >
            <TextInput
              autoCapitalize="none"
              autoCorrect={false}
              keyboardType="email-address"
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              returnKeyType="next"
              securityTextEntry={false}
              style={
                Array [
                  Object {
                    "fontSize": 13,
                    "height": 27,
                    "lineHeight": 24,
                    "paddingBottom": 3,
                  },
                  Object {
                    "color": "#000",
                  },
                  Object {},
                  Object {},
                  Object {},
                ]
              }
              underlineColorAndroid="rgba(0,0,0,0)"
              value=""
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "height": 1,
                  },
                  Object {
                    "backgroundColor": "#E0E0E0",
                  },
                ]
              }
            >
              <View
                collapsable={undefined}
                style={
                  Object {
                    "backgroundColor": "#6d6d6d",
                    "height": 1,
                    "width": 0,
                  }
                }
              />
            </View>
            <Text
              accessible={true}
              allowFontScaling={true}
              collapsable={undefined}
              disabled={false}
              ellipsizeMode="tail"
              onPress={[Function]}
              style={
                Object {
                  "backgroundColor": "rgba(0,0,0,0)",
                  "color": "#9E9E9E",
                  "fontSize": 13,
                  "left": 0,
                  "position": "absolute",
                  "top": 32,
                }
              }
            >
              Email address
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "marginTop": 20,
            }
          }
        >
          <View
            collapsable={undefined}
            style={
              Object {
                "opacity": 0.4,
              }
            }
          >
            <View
              accessibilityComponentType="button"
              accessibilityLabel={undefined}
              accessibilityTraits={
                Array [
                  "button",
                ]
              }
              accessible={true}
              collapsable={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#0084ff",
                  "borderRadius": 4,
                  "elevation": 3,
                  "height": 41,
                  "justifyContent": "center",
                  "opacity": 0.8,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "padding": 8,
                      "textAlign": "center",
                    },
                    Object {
                      "color": "white",
                      "fontSize": 16,
                    },
                  ]
                }
              >
                Resset Password
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
